### PR TITLE
log-backup: allow coordinator to upload GC safepoint

### LINKED ISSUE #40101: log-backup: let the coordinator upload the service safepoint
## Enhancement
For some historical reasons, each TiKV uploads the service GC safepoint by theirselves, which is a little redundant.
We can move it to the coordinator.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #40101 

Problem Summary:
This PR allows the coordinator to push the GC safepoint into PD.

### What is changed and how it works?
This PR added a new interface for `TiKVClusterMeta` named `BlockGCUntil`, which would set a GC safepoint for log backup.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/streamhelper/advancer.go [hunk 1]
```diff
@@ -170,25 +170,31 @@ func tsoBefore(n time.Duration) uint64 {
...
+		})
+		minValue = vsf.Min()
 	})
+	log.Info("[log backup advancer hint] current last region",
+		zap.Stringer("min", minValue), zap.Int("for-polling", len(targets)),
+		zap.String("min-ts", oracle.GetTimeFromTS(minValue.Value).Format(time.RFC3339)))
 	if len(targets) == 0 {
 		c.checkpointsMu.Lock()
 		defer c.checkpointsMu.Unlock()
 		return c.checkpoints.MinValue(), nil
```
Comment (3pointer) [inline br/pkg/streamhelper/advancer.go:196]: return minValue here?

File: br/pkg/streamhelper/advancer.go [hunk 2]
```diff
@@ -390,6 +403,13 @@ func (c *CheckpointAdvancer) importantTick(ctx context.Context) error {
 	if err := c.env.UploadV3GlobalCheckpointForTask(ctx, c.task.Name, c.lastCheckpoint); err != nil {
 		return errors.Annotate(err, "failed to upload global checkpoint")
 	}
+	p, err := c.env.BlockGCUntil(ctx, c.lastCheckpoint-1)
+	if err != nil {
+		return errors.Annotate(err, "failed to update service GC safe point")
+	}
+	if p >= c.lastCheckpoint-1 {
```
Comment (3pointer) [inline br/pkg/streamhelper/advancer.go:410]: maybe record the log whether p is larger or small than  `c.lastCheckpoint-1`.

File: br/pkg/streamhelper/advancer_env.go [hunk 3]
```diff
@@ -32,6 +37,13 @@ type PDRegionScanner struct {
 	pd.Client
 }
 
+// Updates the service GC safe point for the cluster.
+// Returns the latest service GC safe point.
```
Comment (3pointer) [inline br/pkg/streamhelper/advancer_env.go:41]: it seems UpdateServiceGCSafePoint return the minimum service GC safepoint. not the latest?

File: br/pkg/streamhelper/advancer.go [hunk 4]
```diff
@@ -405,10 +400,15 @@ func (c *CheckpointAdvancer) importantTick(ctx context.Context) error {
...
+			"failed to update service GC safe point, current checkpoint is %d, target checkpoint is %d",
+			c.lastCheckpoint-1, p)
+	}
+	if p <= c.lastCheckpoint-1 {
+		log.Info("updated log backup GC safe point.", zap.Uint64("checkpoint", p), zap.Uint64("target", c.lastCheckpoint-1))
 	}
-	if p >= c.lastCheckpoint-1 {
-		log.Info("updated log backup GC safe point.", zap.Uint64("checkpoint", p))
+	if p > c.lastCheckpoint-1 {
+		log.Warn("update log backup GC safe point failed: stale.", zap.Uint64("checkpoint", p), zap.Uint64("target", c.lastCheckpoint-1))
```
Comment (3pointer) [inline br/pkg/streamhelper/advancer.go:413]: when will this happened?

Comment (YuJuncen) [reply br/pkg/streamhelper/advancer.go:413]: When the current checkpoint TS is lower than the GC safepoint (which is rare).

Comment (3pointer) [reply br/pkg/streamhelper/advancer.go:413]: Will it lost data here when the current checkpoint TS is lower than the GC safepoint?

Comment (YuJuncen) [reply br/pkg/streamhelper/advancer.go:413]: Probably. Perhaps we need to issue some fatal error then. 🤔

Comment (YuJuncen) [reply br/pkg/streamhelper/advancer.go:413]: (Maybe in another PR we should introduce TiDB-side fatal errors.)

**Review by 3pointer — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**